Pr 96 fix

sohah opened the pull request:
@Ao-senXiong , thank you for your contributions.

@yannicnoller , This pull request includes changes from the pull request #96, while updating to only reference java 11, while removing everything else, including the reference to the docker setup which we do not have for java11.
With these changes we can accept this pull request.

aosen-xiong committed:
Update gradle api to be compatiable with 7.0

aosen-xiong committed:
Change target to master branch in JPF

aosen-xiong committed:
.gitmodules config update master branch

aosen-xiong committed:
Update to gradle 8.4 and it builds with Java 11

aosen-xiong committed:
Remove files that are the same with Java 11 source code

aosen-xiong committed:
Format

aosen-xiong committed:
CI for this branch

aosen-xiong committed:
Merge branch 'gradle-build-java-11' into compile-java11

aosen-xiong committed:
Fix gralde plugin

aosen-xiong committed:
Build in Java 11

aosen-xiong committed:
Update README.md

aosen-xiong committed:
Update README.md

aosen-xiong committed:
Update README.md

sohah committed:
updating readme, by removing instructions that are not  validated to work on Java 11 version

yannicnoller commented:
Thanks @sohah! Do we know why files were removed?

- jpf-symbc/src/classes/java/awt/image/BufferedImage.java
- jpf-symbc/src/classes/java/awt/image/Kernel.java
- jpf-symbc/src/classes/java/lang/Math.java
- jpf-symbc/src/classes/java/util/Scanner.java

I am not sure whether porting to java11 is a good reason to delete them.

sohah commented:
Hi @yannicnoller , there isn't a very satisifying answser here, I think this needs to be investigated further, hopefully during the runtime exception project, if we get a good proposal for it.
But in general, it looks like java 11 has a different structure, and restriction that prevents us from replacing for example Math.java with our own version, at least in the same way we used to do it. I am seeing that we can do that if we are using `--patch-module` option. Another possibility is to just maintain native peer classes, and intercept bytecodes that we want to symbolically execute, but are not native. This applies to other classes you've mentioned here.

At this point, I think we have two options about this pull request, 
1. either keep that pull request open until we are able to answer these questions and hopefully update the pull request with the necessary changes
2. or accept that pull request and allow others to test what we have so far, perhaps exposing issues that gives us better understanding and clarity. I am okay with either approaches, but perhaps if we choose the second option, then we should have an issue that allows us to track the status of these files going forward.

yannicnoller commented:
@sohah thanks for the update! If we merge, we should flag it as somehow experimental, right now it might look like the official java-11 branch.

sohah committed:
adding a disclaimer about the state of the java 11 branch

sohah commented:
@yannicnoller , I have added a disclamier about this in the readme, and also added a new issue to allows us to track the status of these model classes going forward.

saifk16 commented:
May be these can help here

https://github.com/richieyan/jdk11-patch-module


https://nipafx.dev/five-command-line-options-hack-java-module-system/

saifk16 commented:
> Thanks @sohah! Do we know why files were removed?
> 
> * jpf-symbc/src/classes/java/awt/image/BufferedImage.java
> * jpf-symbc/src/classes/java/awt/image/Kernel.java
> * jpf-symbc/src/classes/java/lang/Math.java
> * jpf-symbc/src/classes/java/util/Scanner.java
> 
> I am not sure whether porting to java11 is a good reason to delete them.

Reference => [PROJECT JIGSAW](https://openjdk.org/projects/jigsaw/)
